desktop/flipper-server-core/src/devices/android/AndroidCertificateProvider.tsx

```diff
@@ -11,14 +11,15 @@ import CertificateProvider from '../../utils/CertificateProvider';
 import {Client} from 'adbkit';
 import * as androidUtil from './androidContainerUtility';
 import {csrFileName, extractAppNameFromCSR} from '../../utils/certificateUtils';
+import {FlipperServerImpl} from '../../FlipperServerImpl';
 
 const logTag = 'AndroidCertificateProvider';
 
 export default class AndroidCertificateProvider extends CertificateProvider {
   name = 'AndroidCertificateProvider';
   medium = 'FS_ACCESS' as const;
 
-  constructor(private adb: Client) {
+  constructor(private flipperServer: FlipperServerImpl, private adb: Client) {
```

Comment:
could we pass in less here than the entire flipperServer? For example just the settings object?

Reply:
I can do that yes 👍 
What's your though about the more suitable thing to pass in?
You are suggesting the settings object, but the minimal one would be the userId. Indeed, the only reason why flipperServer is pass in is to read "this.flipperServer.config.settings.androidUserId"